classes/ObjectHelper.php

```diff
@@ -0,0 +1,201 @@
+            $object->getViewBag()->setProperty('layout', $layout);
+        }
+
+        // Store viewBag data in the object's "settings" property
+        if ($viewBag = array_get($data, 'viewBag')) {
+            $objectData['settings'] = ['viewBag' => $viewBag];
+        }
+
+        // Store allowed properties on the object
+        $fields = ['markup', 'code', 'fileName', 'content', 'itemData', 'name'];
+        foreach ($fields as $field) {
+            if (array_key_exists($field, $data)) {
+                $objectData[$field] = $data[$field];
+            }
+        }
+
+        switch ($type) {
+            case 'page':
+                // Populate the parentFileName for page objects
+                $object->parentFileName = $data['parentFileName'] ?? null;
+
+                $placeholders = array_get($data, 'placeholders');
+                if (is_array($placeholders)) {
+                    $placeholders = array_map([static::class, 'convertLineEndings'], $placeholders);
+                }
+
+                $objectData['placeholders'] = $placeholders;
+                break;
+
+            case 'content':
+                $fileName = $objectData['fileName'];
+
+                if (dirname($fileName) == 'static-pages') {
+                    throw new ApplicationException(Lang::get('winter.pages::lang.content.cant_save_to_dir'));
+                }
+
+                $extension = pathinfo($fileName, PATHINFO_EXTENSION);
+
+                if ($extension === 'htm' || $extension === 'html' || !strlen($extension)) {
+                    $objectData['markup'] = array_get($data, 'markup_html');
+                }
+                break;
+
+            case 'menu':
+                // If no item data is sent through POST, this means the menu is empty
+                if (!isset($objectData['itemData'])) {
+                    $objectData['itemData'] = [];
+                } else {
+                    $objectData['itemData'] = json_decode($objectData['itemData'], true);
+                    if (json_last_error() !== JSON_ERROR_NONE || !is_array($objectData['itemData'])) {
+                        $objectData['itemData'] = [];
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+
+        if (!empty($objectData['markup'])) {
+            $objectData['markup'] = static::convertLineEndings($objectData['markup']);
+        }
+
+        if (!empty($data['objectForceSave']) && $object->mtime) {
+            if ($data['objectMtime'] != $object->mtime) {
+                throw new ApplicationException('mtime-mismatch');
+            }
+        }
+
+        $object->fill($objectData);
+
+        /*
+         * Rehydrate the object viewBag array property where values are sourced.
+         */
+        if ($object instanceof CmsCompoundObject && is_array($viewBag)) {
+            $object->viewBag = $viewBag + $object->viewBag;
+        }
+
+        return $object;
+    }
+}
```

Comment:
Would it make more sense for this to be a trait or behaviour?

Reply:
Given that it's used in a backend controller, a frontend controller, and a backend view partial, probably not.